backend/backend/services/openAI/service.py

```diff
@@ -11,29 +21,82 @@ def generate_prompt(input):
     sample_queries = LIST_OF_EXAMPLES
     return (
         sample_queries
-        + f"""
-Query: Using the above queries, Write a GraphQL query to find: {input}
-Results:"""
+        + PRE_PROMPT.format(input)
     )
 
 
 class OpenAIService:
     def __init__(self, use_prompt=0):
         openai.api_key = OPENAI_API_KEY
 
-    def request_gql_for_graph(self, input_query, subgraph, generic=False):
-        prompt = input_query if generic else generate_prompt(input_query)
-        response = openai.Completion.create(
-            model="text-davinci-003",
-            prompt=generate_prompt(input_query),
-            temperature=0.6,
-            max_tokens=2048,
+    def request_gql_for_graph_llama(self, input_query, subgraph):
+        from backend.services.graph.service import GraphService
+        import regex as re
+        graph_service = GraphService(protocol=subgraph)
+        schema = os.path.join(os.getcwdb().decode("utf-8"), graph_service.subgraph.schema_file_location)
+        mappings = os.path.join(os.getcwdb().decode("utf-8"), "/subgraphs/subgraphs/{}/src/".format(graph_service.subgraph.deployments['base']))
+        examples = os.getcwdb().decode("utf-8")+ "/backend/services/graph/graphql_examples.py"
+        #set recursive = True for case of uniswap etc where there are more sub directories
+        documents = SimpleDirectoryReader(input_files=[examples], recursive=True).load_data()
+        # print("documents", documents)
+        # save to disk
+        # index.save_to_disk('index.json')
+        # load from disk
+        # index = GPTSimpleVectorIndex.load_from_disk('index.json')
+
+        # define LLM
+        llm_predictor = LLMPredictor(llm=OpenAI(temperature=0, model_name="text-davinci-003"))
```

Comment:
is temperature `0` correct?

Reply:
Yes. We dont want to much creativity from. 0 == less, 1 == more creativity